docs/proposals/alternate-api-representations.md

```diff
@@ -0,0 +1,422 @@
+| sv | The server group (`api.k8s.io`) version that should be applied to generic resources returned by this endpoint | Matching server version for the current group and version | If specified, the server should transform generic responses into this version of the server API group. |
+| export | `1` | None | If specified, transform the resource prior to returning to omit defaulted fields. Additional arguments allowed in the query parameter. For legacy reasons, `?export=1` will continue to be supported on the request |
+| pretty | `0`/`1` | `1` | If specified, apply formatting to the returned response that makes the serialization readable (for JSON, use indentation) |
+
+For both export and the more complicated server side `kubectl get` cases, it's likely that
+more parameters are required and should be specified as query parameters. However, the core
+behavior is best represented as a variation on content-type. Supporting both is not limiting
+in the short term as long as we can validate correctly.
+
+
+### Example: Partial metadata retrieval
+
+The client may request to the server to return the list of namespaces as a
+`PartialObjectMetadata` kind, which is an object containing only `ObjectMeta` and
+can be serialized as protobuf or JSON. This is expected to be significantly more
+performant when controllers like the Garbage collector retrieve multiple objects.
+
+    GET /api/v1/namespaces
+    Accept: application/json;g=api.k8s.io,v=v1,as=PartialObjectMetadata, application/json
+
+The server would respond with
+
+    200 OK
+    Content-Type: application/json;g=api.k8s.io,v=v1,as=PartialObjectMetadata
+    {
+      "apiVersion": "api.k8s.io/v1",
+      "kind": "PartialObjectMetadataList",
+      "items": [
+        {
+          "apiVersion": "api.k8s.io/v1",
+          "kind": "PartialObjectMetadata",
+          "metadata": {
+            "name": "foo",
+            "resourceVersion": "10",
+            ...
+          }
+        },
+        ...
+      ]
+    }
+
+Note that the `as` parameter indicates to the server the Kind of the resource, but
+the Kubernetes API convention of returning a List with a known schema continues. An older
+server could ignore the presence of the `as` parameter on the media type and merely return
+a `NamespaceList` and the client would either use the content-type or the object Kind
+to distinguish. Because all responses are expected to be self-describing, an existing
+Kubernetes client would be expected to differentiate on Kind.
+
+An old server, not recognizing these parameters, would respond with:
+
+    200 OK
+    Content-Type: application/json
+    {
+      "apiVersion": "v1",
+      "kind": "NamespaceList",
+      "items": [
+        {
+          "apiVersion": "v1",
+          "kind": "Namespace",
+          "metadata": {
+            "name": "foo",
+            "resourceVersion": "10",
+            ...
+          }
```

Comment:
It will also return other fields of Namespace, not only metadata, right? (I mean, spec and status).